modules/core/src/minmax.cpp

```diff
@@ -1510,28 +1509,39 @@ void cv::minMaxIdx(InputArray _src, double* minVal,
 
     Mat src = _src.getMat(), mask = _mask.getMat();
 
-    if (src.dims <= 2)
+    // HAL call
+    if (src.dims <= 2 || src.isContinuous())
     {
-        CALL_HAL(minMaxIdx, cv_hal_minMaxIdx, src.data, src.step, src.cols*cn, src.rows,
-                 src.depth(), minVal, maxVal, minIdx, maxIdx, mask.data);
-    }
-    else if (src.isContinuous())
-    {
-        int res = cv_hal_minMaxIdx(src.data, 0, (int)src.total()*cn, 1, src.depth(),
-                                   minVal, maxVal, minIdx, maxIdx, mask.data);
+        int srcHalStep, srcHalWidth, srcHalHeight;
+        if (src.dims <= 2)
+        {
+            srcHalStep   = (int)src.step;
+            srcHalWidth  = src.cols * cn;
+            srcHalHeight = src.rows;
+        }
+        else // pass it like one continuous row
+        {
+            srcHalStep   = 0;
+            srcHalWidth  = (int)src.total()*cn;
```

Comment:
To avoid overflow processing should be limited by tensor size.

Reply:
Sorry but I don't understand the issue
HAL function gets continuous array having a size of `dim1 * dim2 * ... * dimN * channels` which is what exactly calculated in this line.